src/mqtt/client.rs

```diff
@@ -921,9 +967,16 @@ impl<'a> EspMqttEvent<'a> {
     }
 
     #[allow(non_upper_case_globals)]
-    pub fn payload(&self) -> EventPayload<'_, EspError> {
+    pub fn payload(&self) -> EventPayload<'_, EspMqttEventErrorWrapper> {
         match self.0.event_id {
-            esp_mqtt_event_id_t_MQTT_EVENT_ERROR => EventPayload::Error(&ERROR), // TODO
+            esp_mqtt_event_id_t_MQTT_EVENT_ERROR => {
+                if let Some(err) = unsafe { self.0.error_handle.as_ref() } {
+                    let wrapped = unsafe { &*(err as *const esp_mqtt_error_codes_t as *const EspMqttEventErrorWrapper) };
+                    EventPayload::Error(wrapped)
+                } else {
+                    todo!()
```

Comment:
Would it be better to use `esp_mqtt_error_type_t_MQTT_ERROR_TYPE_NONE` or create a new value type for `MqttClientError`, such as `ErrorHandleIsNone`?